sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/models/amqp_protocol.hpp

```diff
@@ -12,7 +12,7 @@
 
 struct AMQPVALUE_DECODER_HANDLE_DATA_TAG;
 
-namespace Azure { namespace Core { namespace _internal {
+namespace Azure { namespace Core { namespace _detail {
```

Comment:
I think we should move this to `Azure::Core::Amqp::_detail`. It is a prerogative of the `core-cpp` to declare things in `Azure::Core` namespace. This is `core-amqp-cpp` module, so it should be declaring things in its own sub-namespace. After all, if `core-cpp` declares the same template specialization, there might be problems, and `core-cpp` developer has, in theory, no way of knowing that one if its dependencies declares things in the same scope.

Reply:
I see some other declarations in `Azure::Core::_detail` namespace here, this comment applies to all of them (to all of them who live in `core-ampq-cpp`)

Reply:
I cannot move them to Azure::Core::Amqp::_detail, because this is a specialization of a type declared in the Azure::Core::_detail template. To my knowledge, you cannot declare a specialization of a type in a namespace other than the one in which the type belongs.

Technically the UniqueHandleHelper should remain in the _internal namespace because it's consumed in other packages beyond core, but for packages like Storage::Common and AMQP, there are published types in the _internal namespace, which means the ApiView includes _internal types. 

The point of this is change is that the UniqueHandleHelper template is really an implementation detail for each type owned by the UniqueHandle type. As such, it shouldn't be in the _internal namespace (at a minimum, it forces uAMQP types to be in the publicly documented API surface for AMQP, which is horribly bad.